### PR TITLE
test: use k8 1.14.0-rc.1 by default on all PRs and update k8s libraries

### PR DESCRIPTION
**Pull request by aanm**
1.14.0-rc.1 is a stable enough version that we can start using to test on all PRs also updated the k8s libraries to this version for the same reason.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7462)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: test/provision/manifest/1.14/coredns_deployment.yaml [hunk 1]
```diff
@@ -71,13 +73,19 @@ data:
...
+            max_fails 0
         }
         proxy . /etc/resolv.conf {
-          fail_timeout 10s
-          max_fails 0
+            fail_timeout 10s
+            max_fails 0
         }
+        prometheus :9153
+        forward . /etc/resolv.conf
```
Comment (eloycoto) [inline test/provision/manifest/1.14/coredns_deployment.yaml]: I'm not agree here, there is reasons why I delete this. This is already in the proxy side, and forward can't be declared two times.

File: test/provision/manifest/1.14/coredns_deployment.yaml [hunk 2]
```diff
@@ -71,13 +73,19 @@ data:
...
+        cache 30
```
Comment (eloycoto) [inline test/provision/manifest/1.14/coredns_deployment.yaml]: This cache will make it hard to have in the CI, in these .3 version it behaves different than .2

File: test/test_suite_test.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ import (
 var (
 	log             = logging.DefaultLogger
 	DefaultSettings = map[string]string{
-		"K8S_VERSION": "1.13",
+		"K8S_VERSION": "1.14",
```
Comment (eloycoto) [inline test/test_suite_test.go:42]: also you need to change the test/Vagranfile: 

```
$K8S_VERSION = ENV['K8S_VERSION'] || "1.13"
```

**Review by eloycoto — CHANGES_REQUESTED** [review]

File: test/provision/manifest/1.14/coredns_deployment.yaml (hunk 2 above)
Comment (aanm) [reply test/provision/manifest/1.14/coredns_deployment.yaml]: What does this cache parameter do?

File: test/test_suite_test.go (hunk 3 above)
Comment (aanm) [reply test/test_suite_test.go:42]: thanks. Done!

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: test/provision/manifest/1.14/coredns_deployment.yaml (hunk 2 above)
Comment (eloycoto) [reply test/provision/manifest/1.14/coredns_deployment.yaml]: It keeps the value in the coredns for 30 seconds, but in the version .2 the data was updated if kubernetes watcher send something new, I do not think that in .3 version do that, so the dns can resolve to an invalid name and test will fail. (Happens to me in 1.14 testing)

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
Previous k8s tests were successful https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1846/, fixing k8s 1.14

**Comment by aanm** [comment]
test-me-please  hit https://github.com/cilium/cilium/issues/7373

**Comment by aanm** [comment]
test-me-please failed to provision k8s 1.10

**Comment by aanm** [comment]
test-me-please

**Review by eloycoto — APPROVED** [review]

**Pull request merged**